docs/logging/main-entities.md

```diff
@@ -0,0 +1,91 @@
+---
+id: tofu.logging.main.entities
+title: tofu.logging
+---
+
+
+# tofu.logging core concepts
+
+`tofu.logging` consists of three main things:
+
+## Typeclass `Loggable[A]`
+This is a type class that describes how an arbitrary instance of type A should be presented in the log messages.
+
+## Derivation
+Loggable instances could be derived automatically using `derevo`:
+
+```scala
+import tofu.logging.derivation._
+import derevo.derive
+
+@derive(loggable)
+case class Data(id: Long, weight: Int, name: String)
+```
+
+This annotation puts the instance of `Loggable[Data]` into the (generated if not present) companion object `Data`.
+When the message is logged, the fields of `Data` will be put into the result (e.g. JSON).
+
+### Configured derivation
+
+This derivation can be configured with annotations `hidden`, `masked` and `unembed`:
+```scala
+import tofu.logging.derivation._
+import derevo.derive
+
+@derive(loggable)
+case class ClientData(name: String, surname: String)
+
+@derive(loggable)
+case class Payment(id: Long, @masked(MaskMode.Erase) cardNumber: String, @unembed name: ClientData)
+```
+
+A message `info"This is $payment"` would look like that:
+```json
+{
+  "@timestamp":"2021-08-20T17:13:39.787Z",
+  "loggerName":"some-logger",
+  "threadName":"ioapp-compute-0",
+  "level":"INFO",
+  "message":"This is Payment: Payment{id=3234,cardNumber=...,name=ClientData{name=foo,surname=bar}}",
+  "id":3234,
+  "cardNumber":"...",
+  "name":{
```

Comment:
`@unembed` in case class definition means that here `name` and `surname` should be one layer higher, right?